Update clock dependency

SebMdB opened the pull request:
The data constructors can be useful for testing.

Helkafen commented:
Packdeps found that clock 0.5.1 has been released, but utc.cabal requires the following bounds:

- clock >= 0.3 && < 1
- clock >= 0.3 && < 0.5

Is it ok to update it?

Commit:
update clock dependency

lpeterse commented:
Hi,

I'm a little hesitant with the proposed change. The reason I hide the constructors is to make illegal state unrepresentable. If at all, one might to export them through an `Unsafe` or `Internal` module.

```
Prelude Data.UTC> addYears 25 epoch >>= addDays 23  :: Maybe Date
Just 1995-01-24
```

If this is too tedious to write for testing it might also be a solution to add an `IsString` instance for `Date`, `Time` and `DateTime` that fails with an error on invalid input.

P.S. Thanks for noting the `clock` update. I will merge that one in.

Helkafen commented:
Yes that makes sense. Thanks for your insight.

Commit:
IsString instances for Date, Time and DateTime

Tested with:
- GHC 7.10.3 (stackage lts-5.14)
- GHC 7.8.4 (stackage lts-2.19)

The IsString instances are kept in a separate module,
in order to respect the Safe pragmas.
